client/context/GlobalContext.tsx

```diff
@@ -149,3 +175,50 @@ export default {
     GlobalContextProvider,
     useGlobalContext
 }
+
+/* User Theme */
+
+// Get system theme from CSS media queries
+function systemTheme(): ThemeOption {
+    let defaultTheme = "light" as ThemeOption;
+
+    if (typeof window != "undefined" && typeof window.matchMedia != "undefined") {
```

Comment:
watchMedia is not defined in server-side code, so I had to add the second part of this check to prevent test failures. We can mock the method if we want to actually test this logic (and we should, once it's ready).